### PR TITLE
chore(CPlusPlus): add leaders in arrays

### PR DESCRIPTION
**Pull request by ipsitagoel**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
This is the C++ code to print leaders in an array. If an element is greater than all the elements to its right, the element is a leader.
Language used: C++

**Other information:**
Method: From right to left in the array, scan all elements and record the maximum element until now. When the maximum element changes, print it.
Time Complexity: O(n)

### CODE REVIEW

File: algorithms/CPlusPlus/Arrays/leaders-in-array.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#include <bits/stdc++.h>
+using namespace std;
+
+void findLeader(int a[], int len)
+{
+	int rmax=a[len-1];
+	cout<<rmax;            // The rightmost element is always a leader
+    cout<<endl;
+	for(int i=len-2;i>=0;i--)
+	{
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/leaders-in-array.cpp:15]: fix indentation

File: algorithms/CPlusPlus/Arrays/leaders-in-array.cpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		if (rmax<a[i])
+		{		
+			rmax=a[i];
+			cout<<rmax;
+            cout<<endl;
+		}
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/leaders-in-array.cpp:21]: here also

File: algorithms/CPlusPlus/Arrays/leaders-in-array.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	}
+}
+int main()
+{
+	int n; cin>>n;            // Take the size of the array as input from the user
+    int a[n];
+    for(int i=0;i<n;i++){
+        cin>>a[i];            // Take the elements of the array as input from the user
+    }
+	findLeader(a, n);
```
Comment (aayushjain7) [inline algorithms/CPlusPlus/Arrays/leaders-in-array.cpp:31]: and here too.

**Review by aayushjain7 — CHANGES_REQUESTED** [review]

File: algorithms/CPlusPlus/Arrays/leaders-in-array.cpp (hunk 3 above)
Comment (ipsitagoel) [reply algorithms/CPlusPlus/Arrays/leaders-in-array.cpp]: @aayushjain7 Made the indentation changes

**Review by aayushjain7 — APPROVED** [review]
Great Work! 💯

**Comment by ipsitagoel** [comment]
Thank you!

**Pull request merged**